### PR TITLE
server: --enable-v2 and --enable-v2v3 is decomissioned

### PR DESCRIPTION
**Pull request by serathius**
First part of https://github.com/etcd-io/etcd/issues/12913 for v3.6 release

### CODE REVIEW

**Comment by serathius** [comment]
cc @ptabor

File: tests/e2e/ctl_v3_move_leader_test.go [hunk 1]
```diff
@@ -118,3 +118,14 @@ func testCtlV3MoveLeader(t *testing.T, cfg e2e.EtcdProcessClusterConfig) {
 		}
 	}
 }
+
+func setupEtcdctlTest(t *testing.T, cfg *e2e.EtcdProcessClusterConfig, quorum bool) *e2e.EtcdProcessCluster {
```
Comment (ptabor) [inline tests/e2e/ctl_v3_move_leader_test.go:122]: Is it used solely in this file, or across many ?

If the latter - maybe we should move it to 'common_test.go' file ?

Comment (serathius) [reply tests/e2e/ctl_v3_move_leader_test.go:122]: Only `ctl_v3_move_leader_test.go` uses `setupEtcdctlTest`

**Review by ptabor — APPROVED** [review]
Thank you. Kill it dead.

**Comment by serathius** [comment]
cc @spzala

**Pull request merged**